Nuxt layers support is broken

### UnoCSS version

0.63.0

### Describe the bug

There is a regression after https://github.com/unocss/unocss/pull/4163.
The Uno config is loaded before `uno.config.mjs` is generated:

```bash
ERROR  Cannot start nuxt:  Cannot find module './.nuxt/uno.config.mjs'                                                                  23:03:54
Require stack:
- /home/projects/gqairmgblw.github/uno.config.ts

  Require stack:
  - uno.config.ts
  at Module._resolveFilename (https://gqairmgblwgithub-c5kt.w-credentialless-staticblitz.com/builtins.ddb8d84d.js:144:12334)
  at eval (node_modules/tsx/dist/register-DvZgvjmQ.mjs:33:940)
  at _nodemodule.default._resolveFilename (node_modules/tsx/dist/register-Swi4xIvk.mjs:16:917)
  at Function.resolve (https://gqairmgblwgithub-c5kt.w-credentialless-staticblitz.com/builtins.ddb8d84d.js:163:2873)
  at jitiResolve (node_modules/importx/node_modules/jiti/dist/jiti.cjs#cjs:1:184450)
  at jitiRequire (node_modules/importx/node_modules/jiti/dist/jiti.cjs#cjs:1:186498)
  at import (node_modules/importx/node_modules/jiti/dist/jiti.cjs#cjs:1:196671)
  at eval (uno.config.ts#cjs:1:243)
  at eval_evalModule (node_modules/importx/node_modules/jiti/dist/jiti.cjs#cjs:1:193243)
  at jitiRequire (node_modules/importx/node_modules/jiti/dist/jiti.cjs#cjs:1:187428)
```

We should load Uno config after the `uno.config.mjs` has been generated - https://github.com/unocss/unocss/pull/3838.

### Reproduction

https://stackblitz.com/edit/github-tph5c9?file=nuxt.config.ts,uno.config.ts

### System Info

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/nuxt/src/index.ts
+++ b/packages/nuxt/src/index.ts
@@ -95,31 +95,41 @@ export default mergeConfigs([${configPaths.map((_, index) => `cfg${index}`).join
         },
         write: true,
       })
+      nuxt.hook('app:templatesGenerated', async () => {
+        await loadUnoConfig()
+      })
+    }
+    else {
+      await loadUnoConfig()
     }
 
-    const { config: unoConfig } = await loadConfig(
-      process.cwd(),
-      { configFile: options.configFile },
-      [],
-      options,
-    )
+    async function loadUnoConfig() {
+      const { config: unoConfig } = await loadConfig(
+        process.cwd(),
+        { configFile: options.configFile },
+        [],
+        options,
+      )
 
-    if (
-      isNuxt3()
-      && nuxt.options.builder === '@nuxt/vite-builder'
-      && nuxt.options.postcss.plugins.cssnano
-      && unoConfig.transformers?.some(t => t.name === '@unocss/transformer-directives' && t.enforce !== 'pre')
-    ) {
-      const preset = nuxt.options.postcss.plugins.cssnano.preset
-      nuxt.options.postcss.plugins.cssnano = {
-        preset: [preset?.[0] || 'default', Object.assign(
-          { mergeRules: false, normalizeWhitespace: false, discardComments: false },
-          preset?.[1],
-        )],
+      if (
+        isNuxt3()
+        && nuxt.options.builder === '@nuxt/vite-builder'
+        && nuxt.options.postcss.plugins.cssnano
+        && unoConfig.transformers?.some(t => t.name === '@unocss/transformer-directives' && t.enforce !== 'pre')
+      ) {
+        const preset = nuxt.options.postcss.plugins.cssnano.preset
+        nuxt.options.postcss.plugins.cssnano = {
+          preset: [preset?.[0] || 'default', Object.assign(
+            { mergeRules: false, normalizeWhitespace: false, discardComments: false },
+            preset?.[1],
+          )],
+        }
       }
+      return unoConfig
     }
 
     nuxt.hook('vite:extend', async ({ config }) => {
+      const unoConfig = await loadUnoConfig()
       await nuxt.callHook('unocss:config', unoConfig)
       config.plugins = config.plugins || []
       config.plugins.unshift(...VitePlugin({
@@ -154,6 +164,7 @@ export default mergeConfigs([${configPaths.map((_, index) => `cfg${index}`).join
     }
 
     extendWebpackConfig(async (config) => {
+      const unoConfig = await loadUnoConfig()
       config.plugins = config.plugins || []
       config.plugins.unshift(WebpackPlugin({}, unoConfig))
     })

